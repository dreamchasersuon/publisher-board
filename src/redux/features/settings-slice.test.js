import configureStore from 'redux-mock-store';
import settings, {
  setIsRefresh,
  setTransactions,
  setActiveUser,
} from './settingsFeatureSlice';

describe('redux handling settings actions to set isRefresh', () => {
  test('action set refresh to true', () => {
    const mockStore = configureStore();
    const store = mockStore({});
    store.dispatch(
      setIsRefresh({
        refresh: true,
      }),
    );
    expect(store.getActions()).toMatchSnapshot();
  });

  test('reducer should handle setIsRefresh action', () => {
    expect(
      settings(
        { user: null, isRefresh: false, transactions: [] },
        {
          type: 'settings/setIsRefresh',
          payload: {
            refresh: true,
          },
        },
      ),
    ).toMatchSnapshot();
  });
});

describe('redux handling settings actions to set active user', () => {
  test('action set active user', () => {
    const mockStore = configureStore();
    const store = mockStore({});
    store.dispatch(
      setActiveUser({
        user: {
          user_name: 'Don',
          user_id: 'Huan',
        },
      }),
    );
    expect(store.getActions()).toMatchSnapshot();
  });

  test('reducer should handle setActiveUser action', () => {
    expect(
      settings(
        { user: null, isRefresh: false, transactions: [] },
        {
          type: 'settings/setActiveUser',
          payload: {
            user: {
              user_name: 'Don',
              user_id: 'Huan',
            },
          },
        },
      ),
    ).toMatchSnapshot();
  });
});

describe('redux handling settings actions to set transactions', () => {
  test('action set transactions', () => {
    const mockStore = configureStore();
    const store = mockStore({});
    store.dispatch(
      setTransactions({
        transactions: [
          { id: 1, amount: 3000 },
          { id: 2, amount: 3000 },
        ],
      }),
    );
    expect(store.getActions()).toMatchSnapshot();
  });

  test('reducer should handle setTransactions action', () => {
    expect(
      settings(
        { user: null, isRefresh: false, transactions: [] },
        {
          type: 'settings/setTransactions',
          payload: {
            transactions: [
              { id: 1, amount: 3000 },
              { id: 2, amount: 3000 },
            ],
          },
        },
      ),
    ).toMatchSnapshot();
  });
});
