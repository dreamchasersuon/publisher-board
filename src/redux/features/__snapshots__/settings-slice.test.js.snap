// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redux handling settings actions to set active user action set active user 1`] = `
Array [
  Object {
    "payload": Object {
      "user": Object {
        "user_id": "Huan",
        "user_name": "Don",
      },
    },
    "type": "settings/setActiveUser",
  },
]
`;

exports[`redux handling settings actions to set active user reducer should handle setActiveUser action 1`] = `
Object {
  "isRefresh": false,
  "transactions": Array [],
  "user": Object {
    "user_id": "Huan",
    "user_name": "Don",
  },
}
`;

exports[`redux handling settings actions to set isRefresh action set refresh to true 1`] = `
Array [
  Object {
    "payload": Object {
      "refresh": true,
    },
    "type": "settings/setIsRefresh",
  },
]
`;

exports[`redux handling settings actions to set isRefresh reducer should handle setIsRefresh action 1`] = `
Object {
  "isRefresh": true,
  "transactions": Array [],
  "user": null,
}
`;

exports[`redux handling settings actions to set transactions action set transactions 1`] = `
Array [
  Object {
    "payload": Object {
      "transactions": Array [
        Object {
          "amount": 3000,
          "id": 1,
        },
        Object {
          "amount": 3000,
          "id": 2,
        },
      ],
    },
    "type": "settings/setTransactions",
  },
]
`;

exports[`redux handling settings actions to set transactions reducer should handle setTransactions action 1`] = `
Object {
  "isRefresh": false,
  "transactions": Array [
    Object {
      "amount": 3000,
      "id": 1,
    },
    Object {
      "amount": 3000,
      "id": 2,
    },
  ],
  "user": null,
}
`;
